reporters/GlobalSummaryReporter.js

```diff
@@ -298,18 +294,47 @@ class GlobalSummaryReporter {
       current: this.formatDecimalString(this.totalTokensOutstanding)
     };
 
+    // Get uniswap data via graphql.
+    const uniswapPairDetails = await getUniswapPairDetails(
+      this.web3,
+      this.syntheticContract.address,
+      this.collateralContract.address
+    );
+    const uniswapPairAddress = uniswapPairDetails.pairAddress.toLowerCase();
+    const uniswapClient = getUniswapClient();
+    const allTokenData = (await uniswapClient.request(queries.PAIR_DATA(uniswapPairAddress))).pairs[0];
+    const startPeriodTokenData = (
+      await uniswapClient.request(queries.PAIR_DATA(uniswapPairAddress, this.startBlockNumberForPeriod))
```

Comment:
Do you know if the graph has any sort of rate limiting/overuse protection?

Reply:
No I don't know actually.

Reply:
We aren't running this script super often, so I doubt we need to worry about that now.

Reply:
I can't find any rate-limiting logic in the uniswap.info source, nor do I find mention of it on The Graph website. We should ask the team when we talk with them.